pkg/plugins/plugin.go

```diff
@@ -5,12 +5,14 @@ import (
 )
 
 type VendorPlugin interface {
-	// Return the name of plugin
+	// Name returns the name of plugin
 	Name() string
-	// Return the SpecVersion followed by plugin
+	// Spec returns the SpecVersion followed by plugin
 	Spec() string
-	// Invoked when SriovNetworkNodeState CR is created or updated, return if need dain and/or reboot node
-	OnNodeStateChange(new *sriovnetworkv1.SriovNetworkNodeState) (bool, bool, error)
+	// OnNodeStateChange is invoked when SriovNetworkNodeState CR is created or updated, return if need dain and/or reboot node
+	OnNodeStateChange(*sriovnetworkv1.SriovNetworkNodeState) (bool, bool, error)
 	// Apply config change
 	Apply() error
+	// CheckStatusChanges checks status changes on the SriovNetworkNodeState CR for configured VFs.
+	CheckStatusChanges(*sriovnetworkv1.SriovNetworkNodeState) bool
```

Comment:
cant we use `OnNodeStateChange` to detect that changes are needed ?
perhaps that method need to be extended with another boolean to indicate that changes are required to be performed on the node.

Reply:
that would be convoluted in my opinion. Besides, the logic is different, `OnNodeStateChange` starts from the interfaces of the Status whereas `CheckStatusChange` starts from the interfaces of the Spec. The latter is much simpler and faster.

Reply:
signature is the same, checking logic should be the same IMO
why essentially duplicate the logic and have a bug introduced the minute a PR lands that changes one and not the other ?

Reply:
thx for the detailed information @mlguerrero12, and appologies for delay in reviewing this one.